src/OmniSharp/Api/Refactoring/OmnisharpController.FixUsings.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Simplification;
+using OmniSharp.Models;
+
+namespace OmniSharp
+{
+    public partial class OmnisharpController
+    {
+        //TODO: 
+        //1. Remove unused using statements
+        //2. Sort using statements
+        //3. If possible, replace RemoveRedundantUsings by 
+        //something better from the framework
+        [HttpPost("fixusings")]
+        public async Task<FixUsingsResponse> FixUsings(Request request)
+        {
+            var response = new FixUsingsResponse();
+            var document = _workspace.GetDocument(request.FileName);
+            if (document != null)
+            {
+                var semanticModel = await document.GetSemanticModelAsync();
+                var compilationUnitSyntax = await AddMissingUsings(document, semanticModel, response);
+                await RemoveRedundantUsings(compilationUnitSyntax, response);
+            }
+
+            return response;
+        }
+
+        private async Task<CompilationUnitSyntax> AddMissingUsings(Document document, 
```

Comment:
Logic like this is usually moved into the https://github.com/OmniSharp/omnisharp-roslyn/tree/master/src/OmniSharp/Workers folder. The controllers are a thin layer that expose an end point calling into those workers.